### PR TITLE
Execute example scripts with pytest

### PR DESCRIPTION
**Pull request by hofaflo**
To make sure the example scripts work... if an example should not be tested (e.g. because takes a long time, like the benchmark), it can be excluded by adding its name to `EXCLUDE` in `sleepecg/test/test_examples.py`.

### CODE REVIEW

File: sleepecg/test/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    '*/benchmark/*',
+]
+
+examples_dir = (Path(__file__).parent / '../../examples').resolve()
+
+files_to_test = []
+for pyfile in examples_dir.rglob('*.py'):
+    for pattern in EXCLUDE:
+        if not pyfile.match(pattern):
+            files_to_test.append(pyfile)
```
Comment (cbrnr) [inline sleepecg/test/test_examples.py:23]: The logic doesn't work out - if a file does not match the first pattern you are already adding it to the files to test. But what if it matches another pattern defined in `EXCLUDE`?

Comment (hofaflo) [reply sleepecg/test/test_examples.py]: True, fixed!

**Comment by cbrnr** [comment]
Can you rebase so that we can see if the docs build?

**Comment by hofaflo** [comment]
Done, they do!

**Comment by cbrnr** [comment]
Wow, I'm actually surprised I didn't have this idea earlier! This is super convenient and we don't need another service. Just a quick reminder, RTD is configured to show "stable" (latest release) and "latest" (HEAD), right?

**Comment by hofaflo** [comment]
Right, with "stable" being the default... although the `docs|passing`-badge in the main readme links to "latest". Should we change that?

**Comment by cbrnr** [comment]
Good question. When viewing the README on GitHub we are looking at the "latest" version, so the badge should point to the same version. Therefore, I guess it's OK?

**Comment by hofaflo** [comment]
Okay!

**Comment by cbrnr** [comment]
Thanks @hofaflo!

**Pull request merged**